docs/source-pytorch/advanced/model_parallel.rst

```diff
@@ -67,6 +67,54 @@ For training on unreliable mixed GPUs across the internet check out the :doc:`Hi
 ----
 
 
+************************
+Efficient initialization
+************************
+
+Instantiating a ``nn.Module`` in PyTorch creates all parameters on CPU in float32 precision by default.
+To speed up initialization, you can force PyTorch to create the model directly on the target device and with the desired precision without changing your model code.
+
+.. code-block:: python
+
+    fabric = Trainer(accelerator="cuda", precision="16-true")
+
+    with trainer.init_module():
+        # models created here will be on GPU and in float16
+        model = MyModel()
+
+    trainer.fit(model)
+
+This eliminates the waiting time to transfer the model parameters from the CPU to the device.
+
+When loading a model from a checkpoint, for example when fine-tuning, set `empty_init=True` to avoid expensive
+and redundant memory initialization:
+
+.. code-block:: python
+
+    with trainer.init_module(empty_init=True):
+        # creation of the model is very fast
+        model = MyModel.load_from_checkpoint("my/checkpoint/path.ckpt")
+
+    trainer.fit(model)
+
+For strategies that handle large sharded models (FSDP, DeepSpeed), the :meth:`~lightning.pytorch.trainer.trainer.Trainer.init_module`
+should not be used, instead override the :meth:`~lightning.pytorch.core.hooks.ModelHooks.configure_model` hook:
+
+.. code-block:: python
+
+    class MyModel(LightningModule):
+        def __init__(self):
+            super().__init__()
+            # don't instantiate layers here
+            # move the creation of layers to `configure_model`
+
+        def configure_model(self):
```

Comment:
This name makes sense considering that the previous one was `configure_sharded_model`. I would still suggest a different name because `configure_x` already has a different meaning in Lightning, i.e., the configure_optimizers hook creates the optimizer and returns it. Furthermore, "configuration" refers to something that already exists and gets configured. For these reasons, I'm wondering should we choose a different name? 

The code that goes into this hook is basically what normally is the layer creation in the standard `__init__`. Here are a couple proposals we could think about?

- create_layers
- init_layers (might be confused with weight initializtion???)
- init_model

No strong opinion.

Reply:
I assumed that if `configure_sharded_model` was originally approved, then removing `sharded` would be too :wink:. I'll go with whatever is chosen. My personal thought is that using "mod**el**" is not consistent with "init_mod**ule**"